### ISSUE TITLE
Feature suggestion - make output from spawned executables clearer

### ISSUE THREAD

**Issue by phlummox** [roswell/roswell#486]
`ros` frequently invokes other executables (e.g. using `system()`), and even multiple copies of itself (but with different command-line arguments).

If a problem occurs, this can make it quite hard to track down what has gone wrong.

My suggestion would be:

* When an instance of `ros` starts, the very *first* thing it should do is print a message saying what arguments it was invoked with. (Maybe a `-q|--quiet` flag could be added to suppress this.)
* When running another executable with `system()`, it would be very handy if lines from the output could be prefixed in some way (e.g. with "`> `" or "`+ `", say) to indicate they're *not* being produced by the original `ros` executable directly.

**Comment by snmsts** [comment]
Sounds great but I don't want assign my self to do it.... :(

**Comment by phlummox** [comment]
Indeed. I think it will make solving installation issues harder if it's not added, though. I just did an `strace` of an install to track what executables were being spawned, and it comes to a total of 292 program invocations. Working out what has gone wrong is tricky if you can't tell which of those 292 programs is producing which output.
